src/System.IO.FileSystem/tests/Directory/Exists.cs

```diff
@@ -1,382 +1,287 @@
-
-            Assert.False(result, path);
+            Assert.False(Exists(null));
         }
-    }
 
-    [Fact]
-    public static void Exists_ExistentValidPathAsPath_ReturnsTrue()
-    {
-        var components = IOInputs.GetValidPathComponentNames();
+        [Fact]
+        public void EmptyAsPath_ReturnsFalse()
+        {
+            Assert.False(Exists(string.Empty));
+        }
 
-        foreach (string component in components)
+        [Fact]
+        public void NonExistentValidPath_ReturnsFalse()
         {
-            using (TemporaryDirectory directory = new TemporaryDirectory())
+            Assert.All((IOInputs.GetValidPathComponentNames()), (path) =>
             {
-                string path = Path.Combine(directory.Path, component);
-                Directory.CreateDirectory(path);
-
-                bool result = Directory.Exists(path);
-
-                Assert.True(result, path);
-            }
+                Assert.False(Exists(path), path);
+            });
         }
-    }
-
-
-    [Fact]
-    public static void Exists_NonSignificantWhiteSpaceAsPath_ReturnsFalse()
-    {
-        var paths = IOInputs.GetNonSignificantTrailingWhiteSpace();
 
-        foreach (string path in paths)
+        [Fact]
+        public void ValidPathExists_ReturnsTrue()
         {
-            bool result = Directory.Exists(path);
-
-            Assert.False(result, path);
+            Assert.All((IOInputs.GetValidPathComponentNames()), (component) =>
+            {
+                string path = Path.Combine(TestDirectory, component);
+                DirectoryInfo testDir = Directory.CreateDirectory(path);
+                Assert.True(Exists(path));
+            });
         }
-    }
 
-    [PlatformSpecific(PlatformID.Windows)] // trailing whitespace is significant in filenames
-    [Fact]
-    public static void Exists_ExistingDirectoryWithNonSignificantTrailingWhiteSpaceAsPath_ReturnsTrue()
-    {
-        using (TemporaryDirectory directory = new TemporaryDirectory())
+        [Fact]
+        public void PathWithInvalidCharactersAsPath_ReturnsFalse()
         {
-            var components = IOInputs.GetNonSignificantTrailingWhiteSpace();
-
-            foreach (string component in components)
+            // Checks that errors aren't thrown when calling Exists() on paths with impossible to create characters
+            Assert.All((IOInputs.GetPathsWithInvalidCharacters()), (component) =>
             {
-                string path = directory.Path + component;
-
-                bool result = Directory.Exists(path);
-
-                Assert.True(result, path);
-            }
+                Assert.False(Exists(component));
+            });
         }
-    }
-
-    [Fact]
-    public static void Exists_PathWithInvalidCharactersAsPath_ReturnsFalse()
-    {
-        var paths = IOInputs.GetPathsWithInvalidCharacters();
 
-        foreach (string path in paths)
+        [Fact]
+        public void PathAlreadyExistsAsFile()
```

Comment:
Why are some of the tests have a name that includes expected return values and some don't?